Update the guide to the new format (#1)
hazelcast-guides/springboot-tomcat-session-replication: doc/modules/ROOT/pages/index.adoc

////
This is the base template for Hazelcast integration module guides.

You can clone this repository, work on it and create your guide, and then push to a new repository.
////

> 

Reviewer: ```suggestion
```

> :github-address: [github-repo-address-of-your-guide]

Author: ```suggestion
:github-address: https://github.com/hazelcast-guides/springboot-tomcat-session-replication
```

:hazelcast: Hazelcast IMDG
:framework: Spring Boot

// Use this relative url if you are going to publish the guide on the guides site.
// Note that this url will not work locally and raise asciidoctor errors.
// So, complete the guide with the above url and set the below one just before 
// publishing on the guides site.

> //

Reviewer: ```suggestion
```

:templates-url: templates:ROOT:page$

= Spring Boot Tomcat Session Replication using Hazelcast

> // Content entered directly below the header but before the first section heading is called the preamble.

Reviewer: Please remove the comments from the base guide next time.
```suggestion
```

include::{templates-url}/link-to-repo.adoc[]

== What You’ll Learn

> Spring Boot is a framework that helps to build microservices easily. In a microservice architecture, there are different options to share data among the running services: one of them is caching, which Spring Boot leverages. It also provides session scope data access using built-in Tomcat HTTP sessions. But when running multiple microservices, it quickly becomes a hassle to replicate sessions and share their data across microservices.

Reviewer: I wonder why did you trim the [original content](https://github.com/hazelcast-guides/springboot-tomcat-session-replication-on-kubernetes#spring-boot-tomcat-session-replication-on-kubernetes-using-hazelcast) here.

Author: I can put the original content back. I wasn't sure if that would be too long, I actually liked it and had troubles trimming it.

Reviewer: Let's bring it back. We can divide into multiple categories afterwards.

Author: Alright I will do it

    return config;
}

@Bean
public HazelcastInstance hazelcastInstance(Config hazelcastConfig) {
    return Hazelcast.getOrCreateHazelcastInstance(hazelcastConfig);
}

@Bean
public WebServerFactoryCustomizer<TomcatServletWebServerFactory> customizeTomcat(HazelcastInstance hazelcastInstance) {
    return (factory) -> {
        factory.addContextCustomizers(context -> {
            HazelcastSessionManager manager = new HazelcastSessionManager();
            manager.setSticky(false);
            manager.setHazelcastInstanceName("hazelcastInstance");
            context.setManager(manager);
        });
    };
}
----

- The first bean creates a Hazelcast `Config` object to configure Hazelcast members. We enable the TCI/IP configuration.
- The second bean creates the Hazelcast member using the `hazelcastConfig` bean.
- The third one customizes Tomcat instance in Spring Boot to use Hazelcast Tomcat Session Manager.

Please note that the Hazelcast instance name for `HazelcastSessionManager` object and the instance name for Hazelcast config bean should be the same. Otherwise, `HazelcastSessionManager` wouldn't access the running Hazelcast instance.

== Running the Sample Application

> Build the application by using maven, make sure you are in the project directory

Reviewer: Let's simplify this
```suggestion
In the project directory, build the application:
```

[source,bash]
----

> mvn clean package

Reviewer: It's obviously a shell command, but we've done in the other guides this way
```suggestion
$ mvn clean package
```

----

> Run the application by typing

Reviewer: Seems like no other option than typing :)
```suggestion
Run the application:
```

[source,bash]
----

> java -Dserver.port=8080 -jar target/springboot-tomcat-session-replication-0.1.0.jar 

Reviewer: See above
```suggestion
$ java -Dserver.port=8080 -jar target/springboot-tomcat-session-replication-0.1.0.jar 
```

----

> Open another terminal and rerun the application by typing

Reviewer: ```suggestion
Open another terminal and start the second application:
```

[source,bash]
----
java -Dserver.port=8081 -jar target/springboot-tomcat-session-replication-0.1.0.jar 
----

Now, you can make calls to application endpoints to put data and get it back. Firstly, open another terminal and run the following command to put the data into Hazelcast distributed map:

[source,bash]
----
curl --cookie cookies.txt --cookie-jar cookies.txt -s -L "localhost:8080/put?key=myKey&value=hazelcast"
curl --cookie cookies.txt --cookie-jar cookies.txt -s -L "localhost:8081/get?key=myKey"
----
For each command the output should be like the following:
----
{"value":"hazelcast"}
----

Notice we have to use cookies when testing the application since the data is saved in HTTP sessions.

== Summary

> We created a Spring Boot application that accesses session scope data. To be able to share the session data between each application we used Hazelcast integration. When we  put the key-value pair to our current session as an attribute at port 8080 and we could get the attributes from another port. So we accomplished keeping the session data in each microservice application.

Reviewer: ```suggestion
We created a Spring Boot application that accesses session scope data. To be able to share the session data between each application we used Hazelcast integration. When we  put the key-value pair to our current session as an attribute at port 8080, we could get the attributes from another port. So we accomplished keeping the session data in each microservice application.
```

== See Also

> // Add some links to resources, such as other related guides.
> // Use relative links used on the home page (see https://raw.githubusercontent.com/hazelcast-guides/guides-site/master/home/modules/ROOT/pages/index.adoc)
> 
> - xref:hazelcast-embedded-springboot:ROOT:index.adoc[Hazelcast in SpringBoot]
> - xref:hazelcast-quarkus:ROOT:index.adoc[Hazelcast Client for Quarkus] 

Reviewer: ```suggestion
- xref:hazelcast-embedded-springboot:ROOT:index.adoc[Getting Started with Hazelcast using Spring Boot]
- xref:springboot-session-replication:ROOT:index.adoc[Filter Based Session Replication with Spring Boot and Hazelcast] 
```
